test/functional/p2p_invalid_messages.py

```diff
@@ -42,13 +47,32 @@ def set_test_params(self):
         self.setup_clean_chain = True
 
     def run_test(self):
+        self.test_buffer()
         self.test_magic_bytes()
         self.test_checksum()
         self.test_size()
         self.test_msgtype()
         self.test_large_inv()
         self.test_resource_exhaustion()
 
+    def test_buffer(self):
+        self.log.info("Test message with header split across two buffers, should be received")
+        conn = self.nodes[0].add_p2p_connection(P2PDataStore())
+        # Create valid message
+        msg = conn.build_message(msg_ping(nonce=12345))
+        cut_pos = 12    # Chosen at an arbitrary position within the header
+        # Send message in two pieces
+        before = int(self.nodes[0].getnettotals()['totalbytesrecv'])
+        conn.send_raw_message(msg[:cut_pos])
+        time.sleep(0.01) # Prevent the RPC call from going too early
```

Comment:
I don't think a sleep will consistently avoid races. Any reason why a `sync_with_ping` (on a different connection) or a `wait_until` is not working?

Reply:
Ok, I like `wait_until` better.  More expressive at the very least.  How do you like this?
```suggestion
        # Prevent the RPC call from going too early
        conn.wait_util(int(self.nodes[0].getnettotals()['totalbytesrecv']) != before)
```

Reply:
Should be fine. In case it fails the `assert_equal(middle, before + cut_pos)` should trigger